plugins/module_utils/network/sonic/config/ptp_default_ds/ptp_default_ds.py

```diff
@@ -0,0 +1,439 @@
+    ptp_default_ds_config_path = {
+        'priority1': ptp_default_ds_path + '/priority1',
+        'priority2': ptp_default_ds_path + '/priority2',
+        'domain_number': ptp_default_ds_path + '/domain-number',
+        'log_announce_interval': ptp_default_ds_path + '/ietf-ptp-ext:log-announce-interval',
+        'announce_receipt_timeout': ptp_default_ds_path + '/ietf-ptp-ext:announce-receipt-timeout',
+        'log_sync_interval': ptp_default_ds_path + '/ietf-ptp-ext:log-sync-interval',
+        'log_min_delay_req_interval': ptp_default_ds_path + '/ietf-ptp-ext:log-min-delay-req-interval',
+        'two_step_flag': ptp_default_ds_path + '/two-step-flag',
+        'clock_type': ptp_default_ds_path + '/ietf-ptp-ext:clock-type',
+        'network_transport': ptp_default_ds_path + '/ietf-ptp-ext:network-transport',
+        'unicast_multicast': ptp_default_ds_path + '/ietf-ptp-ext:unicast-multicast',
+        'domain_profile': ptp_default_ds_path + '/ietf-ptp-ext:domain-profile',
+        'source_interface': ptp_default_ds_path + '/ietf-ptp-ext:source-interface',
+    }
+
+    def __init__(self, module):
+        super(Ptp_default_ds, self).__init__(module)
+
+    def get_ptp_default_ds_facts(self):
+        """ Get the 'facts' (the current configuration)
+
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset, self.gather_network_resources)
+        ptp_default_ds_facts = facts['ansible_network_resources'].get('ptp_default_ds')
+        if not ptp_default_ds_facts:
+            return []
+        return ptp_default_ds_facts
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = []
+
+        existing_ptp_default_ds_facts = self.get_ptp_default_ds_facts()
+        commands, requests = self.set_config(existing_ptp_default_ds_facts)
+        if commands:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+
+        changed_ptp_default_ds_facts = self.get_ptp_default_ds_facts()
+
+        result['before'] = existing_ptp_default_ds_facts
+        if result['changed']:
+            result['after'] = changed_ptp_default_ds_facts
+
+        result['commands'] = commands
+
+        new_config = changed_ptp_default_ds_facts
+        old_config = existing_ptp_default_ds_facts
+        if self._module.check_mode:
+            result.pop('after', None)
+            new_config = get_new_config(commands, existing_ptp_default_ds_facts,
+                                        TEST_KEYS_generate_config)
```

Comment:
Could you please try with the default delete operation.
```suggestion
            new_config = get_new_config(commands, existing_ptp_default_ds_facts)
```